sentry/src/routes/channel.rs

```diff
@@ -498,6 +504,68 @@ pub async fn channel_payout<C: Locked + 'static>(
     Ok(Json(SuccessResponse { success: true }))
 }
 
+/// GET `/v5/channel/0xXXX.../get-leaf/spender` request and
+/// GET `/v5/channel/0xXXX.../get-leaf/earner` request
+///
+/// Response: [`GetLeafResponse`]
+pub async fn get_leaf<C: Locked + 'static>(
+    Extension(app): Extension<Arc<Application<C>>>,
+    Extension(channel_context): Extension<ChainOf<Channel>>,
+    Extension(leaf_for): Extension<LeafFor>,
+    Path(params): Path<(ChannelId, Address)>,
+) -> Result<Json<GetLeafResponse>, ResponseError> {
+    let channel = channel_context.context;
+
+    let approve_state = match latest_approve_state(&app.pool, &channel).await? {
+        Some(approve_state) => approve_state,
+        None => {
+            return Err(ResponseError::BadRequest(
+                "No ApproveState message for spender".to_string(),
+            ))
+        }
+    };
+
+    let state_root = approve_state.msg.state_root.clone();
+
+    let new_state = latest_new_state(&app.pool, &channel, &state_root)
+        .await?
+        .ok_or_else(|| ResponseError::BadRequest("No NewState message for spender".to_string()))?;
```

Comment:
This should remain the same. If there is an ApproveState, then there must be a NewState that was approved, so 400 is good.